src/main/java/org/terasology/gf/generator/FoliageProvider.java

```diff
@@ -19,28 +20,29 @@
 @Requires(@Facet(FloraFacet.class))
 public class FoliageProvider implements FacetProvider {
     private float amount;
-    private NoiseTable noise;
+    private WhiteNoise noise;
 
     public FoliageProvider(float amount) {
         this.amount = amount;
     }
 
     @Override
     public void setSeed(long seed) {
-        noise = new NoiseTable(seed + 25873);
+        noise = new WhiteNoise(seed + 25873);
     }
 
     @Override
     public void process(GeneratingRegion region) {
         FoliageFacet facet = new FoliageFacet(region.getRegion(), region.getBorderForFacet(FoliageFacet.class));
         FloraFacet floraFacet = region.getRegionFacet(FloraFacet.class);
+        BlockRegionc foliageRegion = facet.getWorldRegion();
 
-        for (Map.Entry<Vector3i, Float> positionValue : floraFacet.getFlaggedPositions().entrySet()) {
-            Vector3i pos = positionValue.getKey();
+        for (Map.Entry<Vector3ic, Float> positionValue : floraFacet.getWorldEntries().entrySet()) {
+            Vector3ic pos = positionValue.getKey();
             float value = positionValue.getValue();
 
-            if (noise.noise(pos.x, pos.y, pos.z) / 256f < amount) {
-                facet.setFlag(pos, value);
+            if (foliageRegion.contains(pos) && noise.noise(pos.x(), pos.y(), pos.z()) / 256f < amount) {
```

Comment:
Is there no variant to pass a vector to `noise.noise(pos)`? If that is missing, we should add it...

Reply:
No, there isn't. Should that be added to the main `Noise` interface?

Reply:
I think that makes sense, yes. We can use a default implementation for that to only touch that one file.